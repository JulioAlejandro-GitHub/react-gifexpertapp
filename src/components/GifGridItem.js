import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url, id}) => {
    return (
        <div className="card animate__animated animate__tada">
            <img
                src={url} 
                alt={title}
            />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    id: PropTypes.number
}
GifGridItem.defaultProps = {
    url : "https://",
    title : "Default Title",
    id : 1
}