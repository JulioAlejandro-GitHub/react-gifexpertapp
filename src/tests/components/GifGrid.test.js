import React from "react";
import { shallow } from "enzyme";
import {GifGrid} from "../../components/GifGrid";

//llamada a useFetchGifs
import { useFetchGifs } from "../../hooks/useFetchGifs";
// controlar retorno de useFetchGifs
jest.mock("../../hooks/useFetchGifs");

describe('Test <GifGrid />', () => {

    const category = 'Categoria busqueda test';


    test('Test: Componente [GifGrid] toMatchSnapshot()', () => {

        // llama al componente y retorna este valor
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });

        const wrapper = shallow(
            <GifGrid 
                category={category}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    
    test('Test: Hooks [useFetchGifs] carga img :: use [jest.mock]', () => {
        const gifs = [{
            id:1,
            url:'',
            title:''
        },
        {
            id:2,
            url:'',
            title:''
        }];

        // llama al componente y retorna este valor
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });
        const wrapper = shallow( <GifGrid category={category} /> );

        //expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    
});