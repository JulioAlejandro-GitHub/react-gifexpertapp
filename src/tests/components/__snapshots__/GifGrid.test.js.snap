// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> Test: Componente [GifGrid] toMatchSnapshot() 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Categoria busqueda test
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
